src/fs.rs

```diff
@@ -514,14 +519,14 @@ fn directory_listing<S>(
                         body,
                         "<li><a href=\"{}\">{}/</a></li>",
                         file_url,
-                        entry.file_name().to_string_lossy()
+                        file_name
```

Comment:
I think we don't really need to encode file names in directory listing itself?

Reply:
Hi!
Supposing there is a file of which the name is `<p>a.txt` (valid on Linux, at least), then the body of the directory listing is `<body><h1>Index of /dir</h1><ul><li><a href="/dir/<p>a.txt"><p>a.txt</a></li></ul></body>`. 
And the browser will treat the `<p>` as a HTML element. As a result, the webpage is (the browser automatically complements with a closing `</p>`):
![image](https://user-images.githubusercontent.com/8171899/42121952-27355562-7c6c-11e8-9f8e-7655c8824a0a.png)
The correct one should be `<body><h1>Index of /dir</h1><ul><li><a href="/dir/%3Cp%3Ea.txt">&lt;p&gt;a.txt</a></li></ul></body>` so that the `<p>` would not be treated as a HTML element. Result:
![image](https://user-images.githubusercontent.com/8171899/42121979-c2c5a27a-7c6c-11e8-84b8-f6c25d6882be.png).

Reply:
Ah I see, I didn't know you can create such file on Linux.
While it is certainly unlikely, it would be good if special characters would get in.

Is your screenshot from how it looks with your commit?

Reply:
Yes. I tested my commit with [https://github.com/svenstaro/miniserve/](https://github.com/svenstaro/miniserve/) and took that screenshot.